presto-hive/src/main/java/io/prestosql/plugin/hive/HdfsConfigurationInitializer.java

```diff
@@ -84,13 +87,17 @@ public HdfsConfigurationInitializer(HdfsConfig config, Set<ConfigurationInitiali
         this.configurationInitializers = ImmutableSet.copyOf(requireNonNull(configurationInitializers, "configurationInitializers is null"));
     }
 
-    private static Configuration readConfiguration(List<String> resourcePaths)
+    private static Configuration readConfiguration(List<File> resourcePaths)
     {
         Configuration result = new Configuration(false);
 
-        for (String resourcePath : resourcePaths) {
+        for (File resourcePath : resourcePaths) {
+            if (!resourcePath.exists()) {
+                throw new PrestoException(NOT_FOUND, "File does not exist: " + resourcePath);
```

Comment:
Make this `IllegalArgumentException`. We don’t need `PrestoException` since this error isn’t going to a user.

Reply:
then you can `checkArgument(..)`